src/routers/task.js

```diff
@@ -1,95 +1,149 @@
-const express = require('express')
-const Task = require('../models/task')
-const router = new express.Router()
-const auth = require('../middleware/auth')
+const express = require("express");
+const Task = require("../models/task");
+const router = new express.Router();
+const auth = require("../middleware/auth");
+const { uploadImage } = require("../middleware/fileUpload");
+const sharp = require("sharp");
 
-router.post('/tasks', auth, async(req, res) => {
-    const task = new Task({
-        ...req.body,
-        createdBy: req.user._id
-    })
-    try {
-        await task.save()
-        res.status(201).send(task)
-    } catch (error) {
-        res.status(400).send(error)
-    }
-})
+router.post("/tasks", auth, async (req, res) => {
+  const task = new Task({
+    ...req.body,
+    createdBy: req.user._id,
+  });
+  try {
+    await task.save();
+    res.status(201).json(task);
+  } catch (error) {
+    res.status(400).json(error);
+  }
+});
 
 // GET /tasks?completed=???
 // GET /tasks?limit=10&skip=0
 // GET /tasks?sortBy=createdAt:asc
-router.get('/tasks', auth, async(req, res) => {
-    const match = {}
-    const sort = {}
-    if (req.query.completed) {
-        match.completed = req.query.completed === 'true'
-    }
+router.get("/tasks", auth, async (req, res) => {
+  const match = {};
```

Comment:
the same, try catch should wrap whole action

Reply:
Fixed.